robocop/checkers/misc.py

```diff
@@ -1725,3 +1870,83 @@ def check_order(self, node):
                 max_order_indicator = this_node_expected_order
 
     visit_Keyword = visit_TestCase = check_order  # noqa: N815
+
+
+class NonLocalVariableChecker(VisitorChecker):
+    reports = (
+        "no-global-variable",
+        "no-suite-variable",
+        "no-test-variable",
+    )
+
+    def visit_KeywordCall(self, node: KeywordCall):  # noqa: N802
+        keyword_token = node.get_token(Token.KEYWORD)
+
+        if not keyword_token:
+            return
+
+        keyword_name = normalize_robot_name(keyword_token.value, remove_prefix="builtin.")
+
+        if keyword_name == "setglobalvariable":
+            self.report(
+                "no-global-variable",
+                node=keyword_token,
+                lineno=keyword_token.lineno,
+                col=keyword_token.col_offset + 1,
+                end_col=keyword_token.col_offset + len(keyword_token.value) + 1,
+            )
+
+        if keyword_name == "setsuitevariable":
+            self.report(
+                "no-suite-variable",
+                node=keyword_token,
+                lineno=keyword_token.lineno,
+                col=keyword_token.col_offset + 1,
+                end_col=keyword_token.col_offset + len(keyword_token.value) + 1,
+            )
+
+        if keyword_name in ["settestvariable", "settaskvariable"]:
+            self.report(
+                "no-test-variable",
+                node=keyword_token,
+                lineno=keyword_token.lineno,
+                col=keyword_token.col_offset + 1,
+                end_col=keyword_token.col_offset + len(keyword_token.value) + 1,
+            )
+
+    def visit_Var(self, node):  # noqa: N802
+        if ROBOT_VERSION.major < 7:
+            return
+
+        if not node.scope:
+            return
+
+        option_token = node.get_token(Token.OPTION)
+        scope = node.scope.upper()
+
+        if scope == "GLOBAL":
+            self.report(
+                "no-global-variable",
+                node=option_token,
+                lineno=option_token.lineno,
+                col=option_token.col_offset + 1,
+                end_col=option_token.col_offset + len(option_token.value) + 1,
+            )
+
```

Comment:
typically if ``scope`` matches any of those, we can safely return without checking of other conditions as it will never match:

```
if scope == "GLOBAL":
   ...
   return
```